tests/test_pyCombo.py

```diff
@@ -130,3 +130,62 @@ def test_clique_partitioning(test_cp_graph):
     partition, modularity = pycombo.execute(test_cp_graph, treat_as_modularity=True)
     assert modularity == 1.1, modularity
     assert _partitionGroup(partition) == _partitionGroup({0: 0, 1: 0, 2: 1, 3: 2})
+
+
+def test_max_communities(karate):
```

Comment:
Perhaps could be cleaner via
```python
pytest.mark.parametrize("max_comm", [None, 0, -1, -10, 1,  2])
```
and perhaps splitting test into two (one for max_comm checks, one for partitioning. 

PS we tend to use those _partitionGroup or similar code a lot in tests and examples, should we add it as an utility function into the package itself?

Reply:
* I am happy to work on all the above notes, just want to get your thoughts first.

Reply:
Thanks! Yes, everything seems reasonable.

Reply:
cool, setting this as resolved and will take a stab somewhen soon as a separate PR